network/src/commonMain/kotlin/com/wire/kalium/network/session/SessionManager.kt

```diff
@@ -72,6 +76,7 @@ fun HttpClientConfig<*>.installAuth(sessionManager: SessionManager, accessTokenA
                         null
                     }
                 }.also {
+                    sessionManager.afterTokenUpdate()
```

Comment:
Can we `try { } finally { }` this one?

I am a bit worried we could lock the mutex and never release it if some exception is thrown. Doesn't look like a exception could be thrown right now, but maybe the code changes and this is introduced? I am not sure. Just trying to be safe here.